api/scpca_portal/validators.py

```diff
@@ -54,3 +56,89 @@ def validate_project_ids(cls, instance):
                 # TODO: add custom exception
                 raise ValueError(f"Invalid project ID format: {project_id}")
         return instance
+
+
+class DatasetDataResourceExistence:
```

Comment:
```suggestion
class DatasetDataModelRelations
```

Reply:
You could break this up into:
- DatasetDataModelExistence (validates all scpca_ids are present in db)
- DatasetDataModelRelations (validates all samples belong to models)